### ISSUE TITLE
Encoder network choice

### ISSUE THREAD

**Issue by snakers4** [CorentinJ/Real-Time-Voice-Cloning#83]
As far as I can see, you use something in the lines of 3-layer LSTM. Did you try other networks architectures? I am asking, because we are planning to train Tacotron2 + WaveGlow for 30-40 Russian speakers. Originally we just had an idea of slapping and embedding layer in front of both of these networks, but probably adding and encoder would be more cool.

We did a lot of testing of various networks for STT, and roughly came to the following conclusions:
- Ordinary CNN (wav2letter + skip connections) is 2-3x faster than similarly sized LSTM / GRU;
- LSTM is a bit better in quality (but STT is a tougher task than just embedding!);
- A CNN with separable convolutions is 2-3x faster than an ordinary CNN;

TLDR - I was able to speed up fitting a CNN on our dataset from 5-7 days to 1 day on 4 GPUs (to a reasonable plateau level).

Take this with a grain of salt.
Hope this helps.

**Comment by CorentinJ** [comment]
This repo aims to reproduce SV2TTS, where they use a 3 layer LSTM as encoder. That's why I went with that architecture.

The training of the encoder is not quite optimized, as the batch size could be made larger through same-GPU parallelization. Nonetheless, the encoder operates at around 1000x real-time speed, and is the best model for speaker identification I've ever seen, so I see no reason to try and change it.

**Comment by snakers4** [comment]
I see 
Thanks
